pages/hold/request/[id].tsx

```diff
@@ -0,0 +1,340 @@
+
+import type { DiscoveryBibResult } from "../../../src/types/bibTypes"
+import type { DiscoveryItemResult } from "../../../src/types/itemTypes"
+import type { ItemMetadata } from "../../../src/types/itemTypes"
+
+interface HoldRequestPropsType {
+  discoveryBibResult: DiscoveryBibResult
+  discoveryItemResult: DiscoveryItemResult
+  isAuthenticated?: boolean
+}
+
+/**
+ * The Hold Request page renders a form to place a hold on an item.
+ * It renders some basic info about the bib and item based on the bib and item ids in the url,
+ * including pickup location options rendered as radio buttons in the form.
+ */
+export default function HoldRequestPage({
+  discoveryBibResult,
+  discoveryItemResult,
+  isAuthenticated,
+}: HoldRequestPropsType) {
+  const metadataTitle = `Item Request | ${SITE_NAME}`
+
+  const bib = new Bib(discoveryBibResult)
+  const item = new Item(discoveryItemResult, bib)
+  const holdId = `${item.bibId}-${item.id}`
+
+  const [alert, setAlert] = useState(false)
+  const [formPosting, setFormPosting] = useState(false)
+  const notificationRef = useRef<HTMLDivElement>()
+  const { onOpen, setItemMetadata } = useContext(FeedbackContext)
+
+  const router = useRouter()
+  const isLoading = useLoading()
+
+  useEffect(() => {
+    if (alert && notificationRef.current) {
+      notificationRef.current.focus()
+    }
+  }, [alert])
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+
+    try {
+      setFormPosting(true)
+      const response = await fetch(`${BASE_URL}/api/hold/request/${holdId}`, {
+        method: "POST",
+        // TODO: serialize form data
+        body: "",
+      })
+      const responseJson = await response.json()
+
+      if (response.status !== 200) {
+        console.error("Error in hold  request api response", responseJson.error)
+        setAlert(true)
+        setFormPosting(false)
+        return
+      }
+
+      // Success state
+      await router.push(`${PATHS.HOLD_CONFIRMATION}/${holdId}`)
+      setFormPosting(false)
+    } catch (error) {
+      console.error("Error in hold  request api response", error)
+      setAlert(true)
+      setFormPosting(false)
+    }
+  }
+
+  const onContact = (metadata: ItemMetadata) => {
+    setItemMetadata(metadata)
+    onOpen()
+  }
+
+  return (
+    <>
+      <Head>
+        <meta property="og:title" content={metadataTitle} key="og-title" />
+        <meta
+          property="og:site_name"
+          content={metadataTitle}
+          key="og-site-name"
+        />
+        <meta name="twitter:title" content={metadataTitle} key="tw-title" />
+        <title key="main-title">{metadataTitle}</title>
+      </Head>
+      <Layout isAuthenticated={isAuthenticated} activePage="hold">
+        {/* Always render the wrapper element that will display the
+          dynamically rendered notification for focus management */}
+        <Box tabIndex={-1} ref={notificationRef}>
```

Comment:
Nit: rename as `bannerRef` to be slightly more concise, or `bannerContainerRef`?

Reply:
Done!